LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/Mac/LinkADRCmd.cs

```diff
@@ -13,6 +13,11 @@ public class LinkADRCmd : GenericMACCommand
         // private readonly byte dataRateTXPower;
         private readonly byte[] chMask = new byte[2];
 
+        public LinkADRCmd()
+        {
+            this.Length = 2;
+        }
+
         // private readonly byte redondancy;
         public override byte[] ToBytes()
         {
```

Comment:
I would add a `default:` handler. If the cid is not of the type you have cases for this will loop forever. Next question is what to do?
 - throw exception telling that the cid is unknown or not supported?
- skip the loop, log and remain silently?

I prefer the first option.

Reply:
This is correct, I added it to the refactor but this would also make sense here. Thanks for the input